src/foam/dao/CSVSink.js

```diff
@@ -40,22 +70,43 @@ foam.CLASS({
       name: 'isNewLine',
       value: true,
       visibility: 'HIDDEN'
+    },
+    {
+      name: 'axioms',
+      factory: function() {
+        return this.of.getAxiomByName('tableColumns').columns;
+      }
     }
   ],
 
   methods: [
-    function output(value) {
-      if ( ! this.isNewLine ) this.csv += ',';
-      this.isNewLine = false;
-      this.output_(value);
+    {
+      name: 'output',
+      args: [
+        { name: 'value' }
+      ],
+      code: function(value) {
+        if ( ! this.isNewLine ) this.csv += ',';
+        this.isNewLine = false;
+        this.output_(value);
+      },
+      javaCode: `
+        if ( ! getIsNewLine() ) setCsv(getCsv() + ",");
+        setIsNewLine(false);
+        output_(value);
+      `
+
     },
     {
       name: 'output_',
+      args: [
+        { name: 'value' }
+      ],
       code:
         foam.mmethod(
           {
             String: function(value) {
-              this.csv += `"${value.replace(/\"/g, '""')}"`;
+              this.csv += `\"${value.replace(/\"/g, '\"\"')}\"`;
```

Comment:
This change should be unnecessary, I don't think you need to backslash esacpe double quotes inside a backquote in JS.